### ISSUE TITLE
Additional headers on request?

### ISSUE THREAD

**Issue by diniscorreia** [ruby-oembed/ruby-oembed#94]
Hi. Is it possible to send additional headers on the request? 

I needed to add a `Referer` header to get additional information from private videos on Vimeo.

**Comment by metavida** [comment]
No, currently it’s not possible to send any additional headers with requests made by this library. 

Given [Vimeo’s documented need for a header](https://developer.vimeo.com/api/oembed/videos) this seems like a totally fair feature request! Unfortunately I don’t think I’ll have time to implement it in the near future. If you would like to submit a PR that adds a nice API for specifying headers (maybe when configuring an `OEmbed::Provider` perhaps?) I’d be glad to review and try to get it merged in.
